### PR TITLE
give categories a slug

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/2587 by giving categories a `slug` field, and switching to using that rather than names for URL purposes.

Do we want to add this for products, too? /cc @alanmoo

### CODE REVIEW

File: network-api/networkapi/buyersguide/models.py [hunk 1]
```diff
@@ -93,6 +98,10 @@ def published_product_count(self):
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        self.slug = slugify(self.name)
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/models.py]: one risk of doing it this way is that the slug can change if we rename a category, so the URL for that category will change. Alternatively, we set the slug once, at object creation (e.g. `if not self.id`) but then of course the slug will no longer reflect the category itself if we rename it.

I suspect we won't be renaming categories all too often though, so this feels... fine? @cadecairos @alanmoo

**Comment by cadecairos** [comment]
@pomax should we consider falling back to name just in case of links out in the wild?

**Comment by Pomax** [comment]
Good call, let me add that.

File: network-api/networkapi/buyersguide/models.py (hunk 1 above)
Comment (cadecairos) [reply network-api/networkapi/buyersguide/models.py]: that was sort of the thing we agreed upon when this was added to product names, that they wouldn't need to change. We could try to make it only on create, but if we're changing category names infrequently, I think our redirect middleware should be "good enough" to put in a redirect for a category change.

File: network-api/networkapi/buyersguide/views.py [hunk 2]
```diff
@@ -89,7 +89,12 @@ def buyersguide_home(request):
 def category_view(request, categoryname):
```
Comment (cadecairos) [inline network-api/networkapi/buyersguide/views.py]: Can we change the parameter name from `categoryname` to `category_slug` while we're here?

**Pull request merged**